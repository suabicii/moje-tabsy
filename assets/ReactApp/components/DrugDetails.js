import React from "react";
import DosingMoments from "./DosingMoments";

function DrugDetails({drug}) {
    return (
        <>
            <button
                type="button"
                className="btn dropdown-toggle border-0 pt-0 pb-1 px-1 ml-2"
                data-bs-toggle="collapse"
                data-bs-target={`#collapseHours${drug.id}`}
                aria-expanded="false"
                aria-controls={`#collapseHours${drug.id}`}
            >
                Szczegóły
            </button>
            <div className="list-group collapse mt-2" id={`collapseHours${drug.id}`}>
                <h6>Godziny przyjmowania dawek:</h6>
                <ul className="list-group">
                    <DosingMoments drugId={drug.id} content={drug.dosingMoments}
                                   className="list-group-item"/>
                </ul>
            </div>
        </>
    );
}

export default DrugDetails;