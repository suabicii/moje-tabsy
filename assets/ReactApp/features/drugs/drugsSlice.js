import {createSelector, createSlice} from "@reduxjs/toolkit";
import {fetchData} from "../../utils/fetchData";

const drugsSlice = createSlice({
    name: 'drugs',
    initialState: [],
    reducers: {
        addDrug: (state, action) => {
            const {id, name, quantity, quantityMax, unit, dosing, dosingMoments} = action.payload;
            state.push({id, name, quantity, quantityMax, unit, dosing, dosingMoments});
        },
        removeDrug: (state, action) => {
            const id = action.payload;
            return state.filter(drug => drug.id !== id);
        },
        editDrug: (state, action) => {
            const {id, ...updates} = action.payload;
            const drugIndex = state.findIndex(drug => drug.id === id);
            if (drugIndex !== -1) {
                state[drugIndex] = {...state[drugIndex], ...updates};
            }
        },
        setDrugs: (state, action) => action.payload
    },
    sort: true
});

const {reducer, actions} = drugsSlice;
export const {addDrug, removeDrug, editDrug, setDrugs} = actions;

const defaultDrugsSelector = state => state.drugs;

export const sortedDrugsSelector = createSelector(
    defaultDrugsSelector,
    drugs => [...drugs].sort((a, b) => a.name.localeCompare(b.name))
);

export const fetchDrugs = () => async dispatch => {
    await fetchData('/api/drug-list').then(data => {
        dispatch(setDrugs(data));
    }).catch(err => {
        console.log(err);
    });
};

export default reducer;