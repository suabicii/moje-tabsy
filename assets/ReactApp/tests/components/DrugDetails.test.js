import React from "react";
import drugs from "./fixtures/drugs";
import ReactShallowRenderer from "react-test-renderer/shallow";
import DrugDetails from "../../components/DrugDetails";

it('should correctly render DrugDetails component', () => {
    const drug = drugs[0];
    const renderer = new ReactShallowRenderer();

    renderer.render(<DrugDetails drug={drug}/>);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});