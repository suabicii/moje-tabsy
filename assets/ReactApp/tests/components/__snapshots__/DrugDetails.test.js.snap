// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render DrugDetails component 1`] = `
<React.Fragment>
  <button
    aria-controls="#collapseHours1"
    aria-expanded="false"
    className="btn dropdown-toggle border-0 pt-0 pb-1 px-1 ml-2"
    data-bs-target="#collapseHours1"
    data-bs-toggle="collapse"
    type="button"
  >
    Szczegóły
  </button>
  <div
    className="list-group collapse mt-2"
    id="collapseHours1"
  >
    <h6>
      Godziny przyjmowania dawek:
    </h6>
    <ul
      className="list-group"
    >
      <DosingMoments
        className="list-group-item"
        content={
          Object {
            "hour1": "07:00",
            "hour2": "18:00",
          }
        }
        drugId={1}
      />
    </ul>
  </div>
</React.Fragment>
`;
